pages/signin.js

```diff
@@ -122,7 +139,15 @@ class SigninPage extends React.Component {
             />
           </MessageBox>
         )}
-        <SignInOrJoinFree redirect={next || '/'} form={form} routes={this.getRoutes()} />
+        <SignInOrJoinFree
+          redirect={next || '/'}
+          form={form}
+          routes={this.getRoutes()}
+          LoggedInUserTwoFactorAuth={LoggedInUserTwoFactorAuth}
+          submitTOTPCode={values => {
+            this.setState({ TOTPCode: values.TOTPCode });
```

Comment:
Maybe this is where you can just call `await this.props.login(this.props.token, true, values.TOTPCode);`